setup.py

```diff
@@ -13,6 +13,70 @@
 with open(path_version) as f:
     version = f.read().strip()
 
+
+# This function does not handle all corner cases and
+#  assumes a simple requirements.txt is fed to it
+def get_dependencies(requirements_path=None):
+    if not path.exists(requirements_path):
+        return []
+
+    with open(requirements_path) as f:
+        requirements = f.read().splitlines()
+
+    return requirements
+
+
+DEFAULT_REQUIREMENTS = [
+    'colored',
+    'dipy',
+    # h5py is pinned to minor than 3 due to issues with Keras/TF
+    # https://github.com/tensorflow/tensorflow/issues/44467
+    'h5py~=2.10.0',
+    'Keras==2.3.1',
+    'ivadomed',
+    'matplotlib',
+    'nibabel',
+    'numpy',
+    # onnxruntime>=1.5.1 requires `brew install libomp` on macOS.
+    # So, pin to 1.4.0 to avoid having to ask users to install libomp.
+    # ivadomed==2.5.0 would also do this, but #3035 is preventing that.
+    'onnxruntime==1.4.0',
+    'pandas',
+    'psutil',
+    'pyqt5==5.11.3',
+    'pytest',
+    'pytest-cov',
+    'raven',
+    'requests',
+    'requirements-parser',
+    'scipy',
+    'scikit-image',
+    'scikit-learn',
+    'tensorflow~=1.15.0',
+    # PyTorch's Linux/Windows distribution is very large due to its GPU support,
+    # but we only need that for training models. For users, use the CPU-only version
+    # (only available directly from the PyTorch project).
+    # The macOS version has never had GPU support, so doesn't need the workaround.
+    'torch==1.5.0+cpu; sys_platform != "darwin"',
+    'torch==1.5.0; sys_platform == "darwin"',
+    'torchvision==0.6.0+cpu; sys_platform != "darwin"',
+    'torchvision==0.6.0; sys_platform == "darwin"',
+    'xlwt',
+    'tqdm',
+    'transforms3d',
+    'urllib3[secure]',
+    'pytest_console_scripts',
+    'wquantiles',
+]
+
+frozen_dependencies = get_dependencies(path.join(here, 'requirements-freeze.txt'))
+
+dependencies = frozen_dependencies or DEFAULT_REQUIREMENTS
+
+with open(path.join(here, "generated_requirements.txt"), 'wt') as f:
```

Comment:
This is just a nitpick, but what do you think of the name `installed_requirements.txt` instead?

The reason I suggest this is because SCT also generates an install log that gets created by `install_sct`. So, if we choose a similar name, then it might be easier to understand what the file is:

![Screenshot from 2022-01-18 10-47-36](https://user-images.githubusercontent.com/16181459/149970734-e82ceef9-ea23-488e-b7e2-7ede0a4135f2.png)

vs.

![Screenshot from 2022-01-18 10-49-59](https://user-images.githubusercontent.com/16181459/149970972-40ff4068-4cb2-4384-b5e1-1c0e8d8a37c8.png)

Reply:
Yup makes sense